### PR TITLE
[DOCS] Async search: clarify behaviour when submit returns final results 

### LINKED ISSUE #55636: [DOCS] Async search response ID clarification
When an async search successfully completes within the `wait_for_completion_timeout`, and no `keep_alive` parameter is specified such that results will not be available in the cluster, an ID is **not** returned in the response to a submit async search API.

The [documentation for async search APIs](https://www.elastic.co/guide/en/elasticsearch/reference/master/async-search.html) does not currently mention this, but I think it would be good to clarify this behaviour.

### PR DESCRIPTION
**Pull request by javanna**
Closes #55636

### CODE REVIEW

File: docs/reference/search/async-search.asciidoc [hunk 1]
```diff
@@ -98,7 +98,11 @@ some shards returned their results, or when the node that is coordinating the
 
 It is possible to block and wait until the search is completed up to a certain
 timeout by providing the `wait_for_completion_timeout` parameter, which
-defaults to `1` second.
+defaults to `1` second. When the async search completes within such timeout,
+the response won't include the ID as the results are not stored in the cluster.
```
Comment (jimczi) [inline docs/reference/search/async-search.asciidoc:102]: nit: s/the ID/the `id`/

File: docs/reference/search/async-search.asciidoc [hunk 2]
```diff
@@ -98,7 +98,11 @@ some shards returned their results, or when the node that is coordinating the
...
+The `keep_on_completion` parameter, which defaults to `false`, can be set to
+`true` to request that when the async search completes within the
```
Comment (jimczi) [inline docs/reference/search/async-search.asciidoc:104]: nit: s/when the async search completes/when the search completes/ ? It's a parameter for a blocking API so async is maybe misleading.

**Review by jimczi — APPROVED** [review]
I left minor comments

File: docs/reference/search/async-search.asciidoc (hunk 1 above)
Comment (javanna) [reply docs/reference/search/async-search.asciidoc:102]: I think this is how @jrodewig prefers us to mention the id, I will defer to him :)

**Pull request merged**